ddl/column.go

```diff
@@ -145,7 +145,8 @@ func (d *ddl) getColumnInfos(columns []*model.ColumnInfo, positions []*ast.Colum
 				return nil, infoschema.ErrColumnExists.GenByArgs(col.Name)
 			}
 		} else {
-			colInfo, _, err = d.createColumnInfo(tblInfo, col, pos, lastColOffset)
+			// mark sequentially the offset of new column as the last column added idx
+			colInfo, _, err = d.createColumnInfo(tblInfo, col, pos, lastColOffset+idx)
```

Comment:
Is enought this test case in column_change_test.go?
`
+	positions := []*ast.ColumnPosition{
--
  |   | +		{Tp: ast.ColumnPositionNone},
  |   | +		{Tp: ast.ColumnPositionFirst},
  |   | +	}
  |   | +	job := testAddMultipleColumn(c, ctx, d, s.dbInfo, tblInfo, []string{"c3", "c4"}, []interface{}{1, 1}, positions)
  |   | +	c.Assert(errors.ErrorStack(checkErr), Equals, "")
  |   | +	testCheckJobDone(c, d, job, true)
  |   | +	t := testGetTable(c, d, s.dbInfo.ID, tblInfo.ID)
  |   | +	cols := t.Cols()
  |   | +	c.Assert(cols[0].Name.L, Equals, "c4")
  |   | +	c.Assert(cols[3].Name.L, Equals, "c3")

`